### PR TITLE
simplify crumb usage

### PR DESCRIPTION
**Pull request by J02M**
Overly complicated to use crumb, having to duplicate requester parameters, this should make it easier for newcomers to utilize crumb while supporting backwards compatibility.

### CODE REVIEW

**Comment by J02M** [comment]
#705 will help CI pass, we should run codestyle on generated files like eggs:
```
py27 run-test: commands[1] | python -m pycodestyle
./.eggs/pbr-5.2.0-py2.7.egg/pbr/builddoc.py:135:13: E117 over-indented
./.eggs/pbr-5.2.0-py2.7.egg/pbr/git.py:159:29: W605 invalid escape sequence '\*'
./.eggs/pbr-5.2.0-py2.7.egg/pbr/git.py:160:29: W605 invalid escape sequence '\_'
./.eggs/pbr-5.2.0-py2.7.egg/pbr/git.py:161:29: W605 invalid escape sequence '\`'
./.eggs/pbr-5.2.0-py2.7.egg/pbr/packaging.py:306:1: E305 expected 2 blank lines after class or function definition, found 1
./.eggs/pbr-5.2.0-py2.7.egg/pbr/packaging.py:630:1: E305 expected 2 blank lines after class or function definition, found 1
./.eggs/pbr-5.2.0-py2.7.egg/pbr/util.py:211:13: E117 over-indented
./.eggs/pbr-5.2.0-py2.7.egg/pbr/util.py:213:13: E117 over-indented
./.eggs/pbr-5.2.0-py2.7.egg/pbr/tests/test_packaging.py:111:44: W605 invalid escape sequence '\s'
./.eggs/pbr-5.2.0-py2.7.egg/pbr/tests/test_packaging.py:111:48: W605 invalid escape sequence '\s'
./.eggs/pbr-5.2.0-py2.7.egg/pbr/tests/test_packaging.py:111:52: W605 invalid escape sequence '\d'
./.eggs/pbr-5.2.0-py2.7.egg/pbr/tests/test_packaging.py:111:57: W605 invalid escape sequence '\.'
./.eggs/pbr-5.2.0-py2.7.egg/pbr/tests/test_packaging.py:111:61: W605 invalid escape sequence '\d'
./.eggs/pbr-5.2.0-py2.7.egg/pbr/tests/test_packaging.py:111:66: W605 invalid escape sequence '\.'
./.eggs/pbr-5.2.0-py2.7.egg/pbr/tests/test_packaging.py:111:70: W605 invalid escape sequence '\d'
./.eggs/pbr-5.2.0-py2.7.egg/pbr/tests/test_packaging.py:296:24: W605 invalid escape sequence '\*'
./.eggs/pbr-5.2.0-py2.7.egg/pbr/tests/test_packaging.py:303:26: W605 invalid escape sequence '\_'
./.eggs/pbr-5.2.0-py2.7.egg/pbr/tests/test_packaging.py:304:26: W605 invalid escape sequence '\_'
./.eggs/pbr-5.2.0-py2.7.egg/pbr/tests/test_packaging.py:305:28: W605 invalid escape sequence '\_'
./.eggs/pbr-5.2.0-py2.7.egg/pbr/tests/test_packaging.py:312:24: W605 invalid escape sequence '\`'
./.eggs/pbr-5.2.0-py2.7.egg/pbr/tests/test_packaging.py:756:9: E731 do not assign a lambda expression, use a def
./.eggs/pbr-5.2.0-py2.7.egg/pbr/tests/test_setup.py:99:1: E305 expected 2 blank lines after class or function definition, found 1
./.eggs/pbr-5.2.0-py2.7.egg/pbr/tests/test_setup.py:128:1: E305 expected 2 blank lines after class or function definition, found 1
./.eggs/pbr-5.2.0-py2.7.egg/pbr/tests/test_setup.py:165:33: W605 invalid escape sequence '\_'
./.eggs/pbr-5.2.0-py2.7.egg/pbr/tests/test_setup.py:179:32: W605 invalid escape sequence '\_'
./.eggs/pbr-5.2.0-py2.7.egg/pbr/tests/test_wsgi.py:101:39: W605 invalid escape sequence '\d'
./.eggs/pbr-5.2.0-py2.7.egg/pbr/tests/util.py:56:13: E117 over-indented
./.eggs/pbr-5.2.0-py2.7.egg/pbr/tests/util.py:58:13: E117 over-indented
./.eggs/pbr-5.2.0-py2.7.egg/pbr/tests/testpackage/doc/source/conf.py:25:5: E265 block comment should start with '# '
./.eggs/pbr-5.2.0-py2.7.egg/pbr/tests/testpackage/doc/source/conf.py:74:1: E265 block comment should start with '# '
ERROR: InvocationError for command /home/travis/build/pycontribs/jenkinsapi/.tox/py27/bin/python -m pycodestyle (exited with code 1)
```

**Comment by lechat** [comment]
I'd rather not accept this as this will is not backwards compatible and will break code that uses this library.

**Comment by J02M** [comment]
I have now placed the new argument last, this should make the method backwards compatible. Unless I have missed something, in which case I'd like to be enlightened!

File: jenkinsapi/jenkins.py [hunk 1]
```diff
@@ -51,14 +54,23 @@ def __init__(
...
         self.password = password
-        self.requester = requester or Requester(
-            username,
-            password,
-            baseurl=baseurl,
-            ssl_verify=ssl_verify,
-            cert=cert,
-            timeout=timeout
-        )
+        if requester is None:
```
Comment (casz) [inline jenkinsapi/jenkins.py]: the `if requester is None` is the same as `or Requester`

File: jenkinsapi/jenkins.py [hunk 2]
```diff
@@ -51,14 +54,23 @@ def __init__(
...
+            if useCrumb:
```
Comment (casz) [inline jenkinsapi/jenkins.py]: the `if useCrumb` is backward compatible since it defaults to `False`

File: jenkinsapi/jenkins.py [hunk 3]
```diff
@@ -51,14 +54,23 @@ def __init__(
...
+                requester = CrumbRequester
+            else:
```
Comment (casz) [inline jenkinsapi/jenkins.py]: the `else` is the same as `or Requester`

File: jenkinsapi/jenkins.py [hunk 4]
```diff
@@ -51,14 +54,23 @@ def __init__(
...
+
+            self.requester = requester(
+                username,
+                password,
+                baseurl=baseurl,
+                ssl_verify=ssl_verify,
+                cert=cert,
+                timeout=timeout
+            )
+        else:
```
Comment (casz) [inline jenkinsapi/jenkins.py]: the `else` is the as `self.requester = requester`

**Comment by J02M** [comment]
Please review merge request.

**Comment by lechat** [comment]
I think @casz comments in code are valid and make code simpler. Please fix.

Thank you!

**Comment by J02M** [comment]
I'm not sure what you want me to fix, As I read the comments by @casz , they are supporting the backwards compatibility already present in the pull-request.

**Comment by J02M** [comment]
Any review progress?

**Comment by casz** [comment]
@J02M my code comments were used to clarify the code behavior, the code behaves exactly as described in my comments. Hence my approval.

**Pull request merged**